sections/appendix/gibbs_sampling.tex

```diff
@@ -0,0 +1,85 @@
+\subsection{Gibbs Sampling}\label{sec:appendix_gibbs}
+The Gibbs sampling algorithm consists of procedures, in which it is run.
+In the following sections, we explain how these procedures have been implemented.
+\begin{itemize}
+	\item Random Initialization
+	\item Gibbs Sampling
+\end{itemize}
+
+\subsubsection{Random Initialization}
+Before a Gibbs sampling algorithm can run, it need to be randomly initialized with topics.
+To do this, we iterate over each word within each document and assign a random topic for it.
+This procedure is shown in \autoref{lst:randomInit}
+\begin{lstlisting}[language=Python, caption=Random Initialization,label={lst:randomInit}]
+import numpy as np
+
+def rand_initialize(documents: List[np.ndarray]):
+	wt_assignment = []
+	for doc in documents:
+		curr_doc = []
+		for word in doc:
+			# Construct the topic distribution
+			pz = _conditional_distribution()
+			
+			# Draw a new topic and assign it
+			t = np.random.multinomial(1, pz).argmax()
+			curr_doc.append(t)
+			
+			# Increase the topic counts
+			increase_count()
+		wt_assignment.append(curr_doc)
+	return wt_assignment
+\end{lstlisting}
+
+Various parameters are left out to simply the code listing of our initialization method.
+The \emph{\_conditional\_distribution()} function creates a distribution of topics based on the current word.
+It is based on \autoref{eq:gibbs_eq}
+
+\begin{equation}\label{eq:gibbs_eq}
+	\begin{split}
+		P(Z_{d,n} = k |z_{-dn}, \alpha, \lambda) = \\
+		\underbrace{\frac{n_{d,k} + \alpha_{k}}{\sum_{i}^{K} n_{d,i} + \alpha_i}}_{Doc-topic} \times
+		\underbrace{\frac{v_{k, w_{d,n}} + \lambda_{w_{d,n}}}{\sum_{i} v_{k,i} + \lambda_i}}_{Topic-word}
+	\end{split}
+\end{equation}
+where $n_{d,k}$ is the number of times document $d$ uses by topic $k$.
+$v_{k,w}$ is the number of times topic $k$ uses word $w$.
+$\alpha_k$ and $\lambda_w$ is the Dirichlet parameter for the document-topic and topic-word distribution, respectively.
+The doc-topic part of the equation describes how likely topic $t$ appearing in document $d$ and the second part describes how likely which words are most probable in $t$.
+Following the code in \autoref{lst:randomInit}, as we initialize, the words get higher probability of clustering together since, we update the counts every time, we look at a new word in line 16.

```

Comment:
```suggestion
Following the code in \autoref{lst:randomInit}, as we initialize, the words get higher probability of clustering together, since we update the counts every time, we look at a new word in line 16.
```